csrc/host_ir/host_ir.cpp

```diff
@@ -119,6 +119,39 @@ bool PostOnStream::sameAs(const Statement* other) const {
   return false;
 }
 
+LaunchKernel::LaunchKernel(
+    IrBuilderPasskey passkey,
+    int64_t hic_executor_index,
+    std::vector<Val*> inputs,
+    std::vector<Val*> outputs)
+    : Expr(passkey, std::move(inputs), std::move(outputs), {}),
+      hic_executor_index_(hic_executor_index) {}
+
+NVFUSER_DEFINE_CLONE_AND_CREATE(LaunchKernel)
+
+std::string LaunchKernel::toString(int indent_size) const {
+  std::stringstream ss;
+  indent(ss, indent_size) << "LaunchKernel("
+                          << "Inputs: {";
+  std::for_each(inputs().begin(), inputs().end(), [&ss](auto input) {
+    ss << input->toString(0) << ", ";
+  });
+  ss << "}, Outputs: {";
+  std::for_each(outputs().begin(), outputs().end(), [&ss](auto output) {
+    ss << output->toString(0) << ", ";
+  });
+  ss << "})" << std::endl;
+  return ss.str();
+}
+
+int64_t LaunchKernel::getIndex() const {
```

Comment:
Style nit: I'd inline trivial implementation like this in the class declaration so the readers don't need to jump to cpp to understand what the method does.